blog/docs/articles/demystifying-activator-on-path.md

```diff
@@ -0,0 +1,280 @@
+```
+  ...
+  "timestamp": "2023-10-10T15:32:56.178920551Z",
+  "logger": "autoscaler",
+  "caller": "scaling/autoscaler.go:286",
+  "message": "PodCount=0 Total1PodCapacity=10.000 ObsStableValue=1.000 ObsPanicValue=1.000 TargetBC=10.000 ExcessBC=-11.000",
+
+  "timestamp": "2023-10-10T15:32:57.24099875Z",
+  "logger": "autoscaler",
+  "caller": "scaling/autoscaler.go:215",
+  "message": "PANICKING."
+```
+Later on as traffic continues we get proper statistics from activator closer to the rate:
+
+```
+   ...
+   "timestamp":"2023-10-10T15:32:56.949001622Z",
+   "logger":"autoscaler.stats-websocket-server",
+   "message":"Received stat message: {Key:default/autoscale-go-00001 Stat:{PodName:activator-59dff6d45c-9rdxh AverageConcurrentRequests:18.873756322609804 AverageProxiedConcurrentRequests:0 RequestCount:19 ProxiedRequestCount:0 ProcessUptime:0 Timestamp:0}}",
+   "address":":8080"
+
+   ...
+   "timestamp":"2023-10-10T15:32:56.432854252Z",
+   "logger":"autoscaler",
+   "caller":"kpa/kpa.go:188",
+   "message":"Observed pod counts=kpa.podCounts{want:1, ready:0, notReady:1, pending:1, terminating:0}",
+
+   ...
+   "timestamp":"2023-10-10T15:32:57.241052566Z",
+   "logger":"autoscaler",
+   "message":"PodCount=0 Total1PodCapacity=10.000 ObsStableValue=19.874 ObsPanicValue=19.874 TargetBC=10.000 ExcessBC=-30.000",
+```
+Since the pod is not up yet: EBC = 0*10 - floor(19.874) - 10 = -30
+
+Given the new statistics kpa decides to scale to 3 pods at some point.
+
+```
+  "timestamp": "2023-10-10T15:32:57.241421042Z",
+  "logger": "autoscaler",
+  "message": "Scaling from 1 to 3",
+```
+
+But let's see why is this is the case. The log above comes from the multi-scaler which reports a scaled result that contains EBC as reported above and a desired pod count for different windows.
+
+Roughly the final desired number is (there is more logic that covers corner cases and checking against min/max scale limits)
+derived from the dppc we saw earlier.
+
+In this case the target value is 0.7*10=10. So we have for example for the panic window: dppc=ceil(19.874/7)=3
+
+As metrics get stabilized and revision is scaled enough we have:
+
+```
+  "timestamp": "2023-10-10T15:33:01.320912032Z",
+  "logger": "autoscaler",
+  "caller": "kpa/kpa.go:158",
+  "message": "SKS should be in Serve mode: want = 3, ebc = 0, #act's = 2 PA Inactive? = false",
+  
+  ...
+  "logger": "autoscaler",
+  "caller": "scaling/autoscaler.go:286",
+  "message": "PodCount=3 Total1PodCapacity=10.000 ObsStableValue=16.976 ObsPanicValue=15.792 TargetBC=10.000 ExcessBC=4.000",
+```
+
+EBC = 3*10 - floor(15.792) - 10 = 4
+
+Then when we reach the required pod count and metrics are stable we get EBC=3*10 - floor(19.968) - 10=0:
+
+```
+  "timestamp": "2023-10-10T15:33:59.24118625Z",
+  "logger": "autoscaler",
+  "message": "PodCount=3 Total1PodCapacity=10.000 ObsStableValue=19.602 ObsPanicValue=19.968 TargetBC=10.000 ExcessBC=0.000",
+```
+
```

Comment:
```suggestion
Then when we reach the required pod count and metrics are stable we get:

```
  "timestamp": "2023-10-10T15:33:59.24118625Z",
  "logger": "autoscaler",
  "message": "PodCount=3 Total1PodCapacity=10.000 ObsStableValue=19.602 ObsPanicValue=19.968 TargetBC=10.000 ExcessBC=0.000",
```

EBC=3*10 - floor(19.968) - 10=0

```

Reply:
the formatting is weird here due to the multiple blocks, but general idea was to move the EBC calc after the logs, to make the structure the same as the other sections